### PR TITLE
Add shell formatting with shfmt

### PR DESCRIPTION
**Pull request by karlkfi**
- Add make fmt-sh that calls scripts/fmt-bash.sh
- Add EditorConfig with Google-standard shell style config
- Format the existing shell scripts

### CODE REVIEW

**Comment by karlkfi** [comment]
/retest

Unrelated flake:
```
=== CONT  TestRevertClusterRole
    new.go:445: ERROR: timed out waiting for metrics predicate[0]: no results from prometheus query: config_sync_last_sync_timestamp{commit="26b43a286ad411205c11b406205464be9d853d87", component="otel-collector", k8s_pod_name="root-reconciler-65777c464f-w4hmb", status="success"}
```

File: scripts/fmt-bash.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  find scripts e2e -type f \( -name '*.sh' -o -name '*.bash' \)
+)
+mapfile -t check_files < <(
+  echo "${files[@]}" |
+    tr ' ' '\n' |
+    sort |
+    uniq -u
+)
+
+readonly linter=mvdan/shfmt:v3.6.0-alpine
```
Comment (nan-yu) [inline scripts/fmt-bash.sh:29]: nit: rename to `formater`

Comment (karlkfi) [reply scripts/fmt-bash.sh]: done

**Review by nan-yu — APPROVED** [review]
/lgtm

**Comment by karlkfi** [comment]
/retest

[Build log](https://oss.gprow.dev/view/gs/oss-prow-build-kpt-config-sync/pr-logs/pull/GoogleContainerTools_kpt-config-sync/505/kpt-config-sync-presubmit-e2e-multi-repo-test-group3/1641538009629200384) is empty...

**Comment by karlkfi** [comment]
/retest

Build log empty again. Prow node is having issues.

**Pull request merged**